src/data_snack_dynamic_entity/compound_entity/factory.py

```diff
@@ -94,14 +94,14 @@ def _create_compound_entity(
                         f"Source entity {source.entity.__name__} field {e.args[0]} does not exist."
                     )
                 if isinstance(class_field.default, type(MISSING)):
-                    fields.append((entity_field, class_field.type))
+                    fields.add((entity_field, class_field.type))
                 else:
-                    fields_with_defaults.append((entity_field, class_field.type, field(default=class_field.default)))
+                    fields_with_defaults.add((entity_field, class_field.type, field(default=class_field.default)))
 
         entity_template = self._create_compound_entity_template(entity_name=entity_name, sources=sources)
         return make_dataclass(
             entity_name,
-            fields + fields_with_defaults,
+            list(fields) + list(fields_with_defaults),
```

Comment:
Maybe `list(fields | fields_with_defaults)`?

Reply:
Here also fields order is important, with your suggestion I get `TypeError: non-default argument '' follows default argument`